prow/tide/tide.go

```diff
@@ -234,6 +237,38 @@ func accumulate(presubmits []string, prs []pullRequest, pjs []kube.ProwJob) (suc
 	return
 }
 
+func prNumbers(prs []pullRequest) []int {
+	var nums []int
+	for _, pr := range prs {
+		nums = append(nums, int(pr.Number))
+	}
+	return nums
+}
+
+func (c *Controller) pickBatch(sp subpool) ([]pullRequest, error) {
+	r, err := c.gc.Clone(sp.org + "/" + sp.repo)
+	if err != nil {
+		return nil, err
+	}
+	defer r.Clean()
```

Comment:
TODO: don't erase the whole repo

Reply:
It's fine, the git client keeps a mirror clone that we checkout from. A git checkout from a local mirror clone is fast.